### PR TITLE
Development environment setup

### PR DESCRIPTION
**Pull request by anicolao**
This set of changes makes it so that:

- we can switch the codebase to typescript
- we can use separate files for different functionality
- you may no longer check in unformatted code or code that isn't lint free
- you must use `npm` to install development tools and produce a working extension
- there is documentation on how to get started developing

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,2 +1,4 @@
 *~
-*.zip
\ No newline at end of file
+*.zip
+dist
+node_modules
```
Comment (Tsangares) [inline .gitignore:4]: I would prefer using a trailing slash in got ignore to refer to directories when we know they are directories.

```
node_modules/
```

**Review by Tsangares — CHANGES_REQUESTED** [review]

File: .husky/pre-commit [hunk 2]
```diff
@@ -0,0 +1,5 @@
+#!/usr/bin/env sh
```
Comment (Tsangares) [inline .husky/pre-commit:1]: Excellent use of environment thank you!

File: source/cookie.js [hunk 3]
```diff
@@ -1,2 +1,95 @@
...
+!(function (e, t) {
```
Comment (Tsangares) [inline source/cookie.js:2]: Please delete and remove this file and update the manifest to not require cookies as a dependency.

**Review by Tsangares — CHANGES_REQUESTED** [review]
Small changes, then I will pull tomorrow. Great work! Webpack will make this project a lot better and make approvals slower XD

The price you pay for organization.

File: source/cookie.js (hunk 3 above)
Comment (Tsangares) [reply source/cookie.js:2]: Any reference to the cookie lib just comment out and I willviodatw layer to plugin storage. The only code suing cookie is an auto merge apikey which has a bug with warp gates that needs to be fixed before integration.

Comment (anicolao) [reply source/cookie.js:2]: Can you do this as a different commit? It has nothing to do with webpack. I can rebase the branch on main once you've committed the fix.

**Comment by Tsangares** [comment]
It looks like the code for caching was revised in this commit so thats why there are merge conflicts.

**Comment by Tsangares** [comment]
Oh i see there was also a large syntax change. Not totally sure which one is correct but they are the same code functionally.

**Comment by Tsangares** [comment]
Yeah this thing needs a merge. These two intel files are totally different in a ton of small ways.

**Pull request merged**